### PR TITLE
[buildkite] Upload build artifact and add to build scan

### PR DESCRIPTION
**Pull request by brianseeders**
- Upload build artifact to Buildkite.
- Link to the artifacts from the buildscan
- Add a couple of extra buildscan values

### CODE REVIEW

File: build-tools-internal/src/main/groovy/elasticsearch.build-complete.gradle [hunk 1]
```diff
@@ -64,5 +65,27 @@ if (buildNumber && performanceTest == null && GradleUtils.isIncludedBuild(projec
...
+    if (uploadFile.exists() && System.getenv("BUILDKITE") == "true") {
+      try {
+        println "Uploading buildkite artifact: ${uploadFilePath}..."
+        new ProcessBuilder("buildkite-agent", "artifact", "upload", uploadFilePath)
+          .start()
+          .waitFor()
+
+        println "Generating buildscan link for artifact..."
+
+        def process = new ProcessBuilder("buildkite-agent", "artifact", "search", uploadFilePath, "--step", System.getenv('BUILDKITE_JOB_ID'), "--format", "%i").start()
```
Comment (mark-vieira) [inline build-tools-internal/src/main/groovy/elasticsearch.build-complete.gradle:78]: It's very annoying that `artifact upload` doesn't just return the id and we have to do a separate query here.

File: build-tools-internal/src/main/groovy/elasticsearch.build-scan.gradle [hunk 2]
```diff
@@ -101,7 +101,8 @@ buildScan {
       tag 'CI'
       link 'CI Build', buildKiteUrl
       value 'Job Number', System.getenv('BUILDKITE_BUILD_NUMBER')
-
+      value 'Pipeline', System.getenv('BUILDKITE_PIPELINE_SLUG')
+      value 'Job Name', System.getenv('BUILDKITE_LABEL')
```
Comment (mark-vieira) [inline build-tools-internal/src/main/groovy/elasticsearch.build-scan.gradle:105]: We should think about sanitizing the label to produce a "slug like" value that's easier to search on. Also stripping out things like emojis (we use them in serverless).

File: build-tools-internal/src/main/groovy/elasticsearch.build-scan.gradle [hunk 3]
```diff
@@ -101,7 +101,8 @@ buildScan {
       tag 'CI'
       link 'CI Build', buildKiteUrl
       value 'Job Number', System.getenv('BUILDKITE_BUILD_NUMBER')
-
+      value 'Pipeline', System.getenv('BUILDKITE_PIPELINE_SLUG')
```
Comment (mark-vieira) [inline build-tools-internal/src/main/groovy/elasticsearch.build-scan.gradle:104]: Can we add these as tags too? We do that for Jenkins.

**Comment by brianseeders** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|8.11|[](https://github.com/elastic/elasticsearch/pull/100859)|
|✅|8.10|[](https://github.com/elastic/elasticsearch/pull/100860)|
|✅|7.17|[](https://github.com/elastic/elasticsearch/pull/100861)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**